modules/apps/segments/segments-context-vocabulary/src/main/java/com/liferay/segments/context/vocabulary/internal/configuration/persistence/listener/SegmentsContextVocabularyConfigurationModelListener.java

```diff
@@ -56,16 +60,19 @@ public void onBeforeSave(String pid, Dictionary<String, Object> properties)
 				ResourceBundleUtil.getString(
 					_getResourceBundle(),
 					"please-enter-a-valid-session-property-name"),
-				SegmentsContextVocabularyConfiguration.class, getClass(),
+				SegmentsContextVocabularyCompanyConfiguration.class, getClass(),
 				properties);
 		}
 
-		if (_isDefined(pid, entityField)) {
+		if (_isDefined(
+				String.valueOf(properties.get("assetVocabulary")), entityField,
+				String.valueOf(properties.get("companyId")))) {
+
 			throw new DuplicatedSegmentsContextVocabularyConfigurationModelListenerException(
 				ResourceBundleUtil.getString(
 					_getResourceBundle(),
 					"this-field-is-already-linked-to-one-vocabulary"),
-				SegmentsContextVocabularyConfiguration.class, getClass(),
+				SegmentsContextVocabularyCompanyConfiguration.class, getClass(),
```

Comment:
Why do we use SegmentsContextVocabularyCompanyConfiguration, I mean could it be duplicated in SegmentsContextVocabularyConfiguration too, right?